posthog-react-native/src/posthog-rn.ts

```diff
@@ -112,7 +66,13 @@ export class PostHog extends PostHogCore {
           this._semiAsyncStorage?.setItem(PostHogPersistedProperty.AnonymousId, legacyValues.anonymousId)
         }
       }
+    }
 
+    this._initPromise = setupAsync()
+
+    // Not everything needs to be in the init promise
+    this._initPromise.then(async () => {
+      this._isInitialized = true
```

Comment:
This flag is being flipped but not being used anywhere, do we need it?
Either we always call `await this._initPromise` to be sure the promise is resolved or we check the `_isInitialized` flag, but right now seems not needed.

Reply:
nvm its used in the core project